src/iterator.vala

```diff
@@ -58,37 +48,26 @@ public class Vast.Iterator : Object
     public Value
     get_value ()
     {
-        return array.get_value (_cursor);
+        return memory_to_value(get(), array.scalar_type, array.scalar_size);
     }
 
     public void
-    set (void* val)
+    set (void* memory)
     {
-        Memory.copy ((uint8*) array.data.get_data () + _offset, val, array.scalar_size);
+        Memory.copy (get(), memory, array.scalar_size);
```

Comment:
Looks like if the array exposes an internal void * / uint8* member that is data.get_data(), then we can avoid all of these functions.